### PR TITLE
add ndt_pcl_modified readme

### PR DESCRIPTION
**Pull request by YamatoAndo**
## Description
Add README.md for ndt_pcl_modified

## Review Procedure

Check README.md.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+## Purpose
+
+This package is a modified version of NDT library, which is implemented in Point Cloud Library (https://github.com/PointCloudLibrary/pcl).
+
+
+
+## Change Points
+  - You can get the Hessian matrix by getHessian(). 
+  - You can get the estimated position for each iteraion by getFinalTransformationArray().
```
Comment (yukkysaito) [inline localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md:11]:
```suggestion
  - You can get the estimated position for each iteration by getFinalTransformationArray().
```

Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md]: @yukkysaito Thank you!!

File: localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+## Purpose
+
+This package is a modified version of NDT library, which is implemented in Point Cloud Library (https://github.com/PointCloudLibrary/pcl).
+
+
+
+## Change Points
+  - You can get the Hessian matrix by getHessian(). 
+  - You can get the estimated position for each iteration by getFinalTransformationArray().
+  - It don't align the 6 axes at the same time, just process the rotational component first, then process the 6 axes. [experimental feature]
```
Comment (kmiya) [inline localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md:12]:
```suggestion
  - It doesn't align the 6 axes at the same time, just processes the rotational component first, then processes the 6 axes. [experimental feature]
```
Nit

Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md]: @kmiya Thank you!!

File: localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# ndt_pcl_modified
+
+## Purpose
+
+This package is a modified version of NDT library, which is implemented in Point Cloud Library (https://github.com/PointCloudLibrary/pcl).
```
[the remark this answers is not in the document]
Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md]: @IshitaTakeshi Thank you!! https://github.com/tier4/AutowareArchitectureProposal.iv/pull/599/commits/e584ff41c0c883837a337096e47d3247cc704d7e

File: localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md]: @IshitaTakeshi Thank you!! https://github.com/tier4/AutowareArchitectureProposal.iv/pull/599/commits/e584ff41c0c883837a337096e47d3247cc704d7e

**Review by IshitaTakeshi — APPROVED** [review]

File: localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# ndt_pcl_modified
+
+## Purpose
+
+This is a modification of [PCL](https://github.com/PointCloudLibrary/pcl)'s NDT.
+
+## Change Points
```
Comment (IshitaTakeshi) [inline localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md:7]: "Modifications" would be more fluent but I think "change points" makes sense anyway

Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_pcl_modified/README.md]: @IshitaTakeshi Thank you!! https://github.com/tier4/AutowareArchitectureProposal.iv/pull/599/commits/71bcc001037b61da5dc1c471f8cae637bfb51749

**Pull request merged**